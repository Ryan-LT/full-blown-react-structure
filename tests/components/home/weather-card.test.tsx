import React from 'react';
import { WeatherCard } from '@components';
import { cleanup, render } from '../../renderTest';
import '@testing-library/jest-dom';

afterEach(cleanup);

describe('<WeatherCard />', () => {
  it('should take a snapshot', () => {
    const testProps = {
      applicable_date: 'Fri Apr 02 2021 17:18:04 GMT+0700',
      min_temp: 16,
      max_temp: 23,
    };
    const { asFragment } = render(<WeatherCard {...testProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render without issue', () => {
    const testProps = {
      applicable_date: new Date().toString(),
      min_temp: 16,
      max_temp: 23,
    };
    const component = render(<WeatherCard {...testProps} />);
    expect(component).toBeTruthy();
  });

  it('should have max temp', () => {
    const testProps = {
      applicable_date: new Date().toString(),
      min_temp: 16,
      max_temp: 23,
    };
    const { getByTestId } = render(<WeatherCard {...testProps} />);
    expect(getByTestId('max-temp')).toHaveTextContent('23');
  });

  it('should have min temp', () => {
    const testProps = {
      applicable_date: new Date().toString(),
      min_temp: 16,
      max_temp: 23,
    };
    const { getByTestId } = render(<WeatherCard {...testProps} />);
    expect(getByTestId('min-temp')).toHaveTextContent('16');
  });

  it('should show Friday', () => {
    const testProps = {
      applicable_date: 'Fri Apr 02 2021 17:18:04 GMT+0700',
      min_temp: 16,
      max_temp: 23,
    };
    const { getByTestId } = render(<WeatherCard {...testProps} />);
    expect(getByTestId('card-day')).toHaveTextContent('Friday');
  });
});
