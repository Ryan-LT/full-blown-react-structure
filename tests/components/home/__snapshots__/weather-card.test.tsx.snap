// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherCard /> should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="weather-card"
  >
    <p
      data-testid="card-day"
    >
      Friday
    </p>
    <p
      data-testid="max-temp"
    >
      Max: 23.0° C
    </p>
    <p
      data-testid="min-temp"
    >
      Min: 16.0° C
    </p>
  </div>
</DocumentFragment>
`;
